src/main/kotlin/team/msg/hiv2/global/security/token/GenerateJwtAdapter.kt

```diff
@@ -0,0 +1,52 @@
+package team.msg.hiv2.global.security.token
+
+import io.jsonwebtoken.Jwts
+import io.jsonwebtoken.SignatureAlgorithm
+import org.springframework.stereotype.Component
+import team.msg.hiv2.domain.auth.persistence.entity.RefreshTokenEntity
+import team.msg.hiv2.domain.auth.persistence.repository.RefreshTokenRepository
+import team.msg.hiv2.domain.auth.presentation.data.response.TokenResponse
+import team.msg.hiv2.global.security.spi.GenerateJwtPort
+import java.security.Key
+import java.time.LocalDateTime
+import java.util.*
+
+@Component
+class GenerateJwtAdapter(
+    private val jwtProperties: JwtProperties,
+    private val refreshTokenRepository: RefreshTokenRepository
+) : GenerateJwtPort{
+
+    override fun generate(email: String): TokenResponse {
+        val accessToken = generateAccessToken(email, jwtProperties.accessSecret)
+        val refreshToken = generateRefreshToken(email, jwtProperties.refreshSecret)
+        val accessExpiredAt = getAccessTokenExpiredAt()
+        val refreshExpiredAt = getRefreshTokenExpiredAt()
+        refreshTokenRepository.save(RefreshTokenEntity(refreshToken, email, jwtProperties.refreshExp))
+        return TokenResponse(accessToken, refreshToken, accessExpiredAt, refreshExpiredAt)
+    }
+
+    private fun generateAccessToken(email: String, secret: Key): String =
+        Jwts.builder()
+            .signWith(secret, SignatureAlgorithm.HS256)
+            .setSubject(email)
+            .claim("type", JwtProperties.accessType)
+            .setIssuedAt(Date())
+            .setExpiration(Date(System.currentTimeMillis() + jwtProperties.accessExp * 1000))
+            .compact()
+
+    private fun generateRefreshToken(email: String, secret: Key): String =
+        Jwts.builder()
+            .signWith(secret, SignatureAlgorithm.HS256)
+            .setSubject(email)
+            .claim("type", JwtProperties.refreshType)
+            .setIssuedAt(Date())
+            .setExpiration(Date(System.currentTimeMillis() + jwtProperties.accessExp * 1000))
```

Comment:
```suggestion
            .setExpiration(Date(System.currentTimeMillis() + jwtProperties.refreshExp * 1000))
```

액세스의 만료말고 refresh의 만료를 가져와야할거같습니다